godal.go

```diff
@@ -3365,6 +3365,222 @@ func (ds *Dataset) LayerByName(name string) *Layer {
 	return &Layer{majorObject{C.GDALMajorObjectH(hndl)}}
 }
 
+type SpatialFilter struct {
+	geom *Geometry
+}
+
+func (sf SpatialFilter) setExecuteSQLOpt(eso *executeSQLOpts) {
+	eso.spatialFilter = sf
+}
+
+func NewSpatialFilter(geom *Geometry) SpatialFilter {
+	return SpatialFilter{geom}
+}
```

Comment:
The API will be better worded with
```suggestion
type SpatialFilterOption struct {
	geom *Geometry
}

func (sf SpatialFilterOption) setExecuteSQLOpt(eso *executeSQLOpts) {
	eso.spatialFilter = sf
}

func SpatialFilter(geom *Geometry) SpatialFilterOption {
	return SpatialFilterOption{geom}
}
```